### ISSUE TITLE
Pipe external request to response stream. 

### ISSUE THREAD

**Issue by ottiker** [jillix/wrabbit#5]
Instead of downloading the complete data to the memory, use the Stream API to pipe the external file to the HTTP response.

**Comment by IonicaBizau** [comment]
How is that possible since we need to add the wrapping lines?

**Comment by ottiker** [comment]
HTTP's `request` and `response` objects are actual [Streams](https://nodejs.org/api/stream.html#stream_api_for_stream_consumers). So it's possible to listen to the `data` event and change the data, before it's written to the `response` stream.

**Comment by IonicaBizau** [comment]
So, if I got it correctly we would have:
- if wrap
  - `res.write("wrapping line")`
- pipe tinyreq to response
- if wrap (will this work?)
  - `res.write("return module; });");`

**Comment by ottiker** [comment]
This is just raw thinking, but something like this could work:

``` js
var headerComplete = false;
var firstLineWritten = false;

extRes.on('data', function (chunk) {
    if (!headerComplete) {
        // ..collect header
        if (/*header is complete*/) {
            headerComplete = true
            link.res.writeHead(200, {/*collected header*/});
        }
        return;
    }

    if (!firstLineWritten) {
        firstLineWritten = true;
        link.res.wirte('first line wrapping code');
    }

    link.res.write(chunk);
});

extRes.on('end', function () {
    link.res.end('last line wrapping code');
});
```

**Comment by ottiker** [comment]
And if there's no wrapping just use the `pipe` method to connect the two streams.

**Comment by IonicaBizau** [comment]
Fixed via #6.
